### ISSUE TITLE
PdfFileMerger generates blank pages on Python 3 when given file handlers

### ISSUE THREAD

**Issue by mluszczyk** [mstamy2/PyPDF2#293]
I'm using the merger like in the snippet below. Whatever input PDF file I provide, I get a PDF with the same number of pages, but all of them are blank (even if in the input file they are not blank).

``` python
from PyPDF2 import PdfFileMerger

merger = PdfFileMerger()

INPUT_PATH = "/tmp/a-pdf-file.pdf"
OUTPUT_PATH = "/tmp/output.pdf"

with open(INPUT_PATH, 'rb') as input_file:
    merger.append(input_file)

with open(OUTPUT_PATH, 'wb') as out:
    merger.write(out)
```

Does not happen on python 2.7, but happens in exactly the same environment on python 3.5. Tested on `PyPDF2==1.26.0`.

One finding I have is that the `if` branch on line [116](https://github.com/mstamy2/PyPDF2/blob/master/PyPDF2/merger.py#L116) is never entered in Python 3, but it is entered on Python 2. The reason is that `isinstance(open("/tmp/x"), file)` is evaluated to `True` on Python 2, but `isinstance(open("/tmp/x"), io.FileIO)` is evaluated to `False` on Python 3 (note `from io import FileIO as file` at the top of the file).

**Comment by keliomer** [comment]
@mstamy2 
Seems the fix (#294 ) hasn't been made available on the version available through PyPI. 
Experiencing the same issue as #293 with Python 3.6.3.

I uninstalled and reinstalled through pip, and  still ended up having to manually changing the line @mluszczyk did in their PR.

Could we reopen this issue until the fix is live and pip-able?

**Comment by cgarbin** [comment]
Passing along a solution I got from https://stackoverflow.com/a/49927541/336802: keep the input files open while writing the output file.

This works with the currently released version of PyPDF2.
